### DISCUSSION TITLE
Some questions for using Nextspace

### DISCUSSION THREAD

**Discussion by mncho** [trunkmaster/nextspace#429] [Q&A]
Hello, I have recently moved from XFCE to WindowMaker a few months ago, but today decided to make the full leap to a full GNUStep desktop enviroment, rather than continue to use WindowMaker just as a WM. Also, I want to potentially contribute but may need to learn about Objective-C first, and want to daily drive this for a while. I had no problems with install, but am very unfamiliar with many things, and need some help.

1. Why does the launcher for me not actually launch anything? The run option is grayed out and I cant figure out how to get it to work.
2. Is there any file which will autostart apps when the WM loads? I'd like to run compton (screen tearing is very unpleasant to me), but I cant get it to start automatically.
3. How to change the image/name of icons? I use brave browser, and I cannot figure out how to get it to show with a logo, rather than be the 'unknown program' abd called 'Brave-browser.'
4. I cannot launch anything besides .app directories via the File Viewer, I seem to need to launch everything via the terminal or dock. is there a way to create .app that just will launch a regular program, so I do not have to use the terminal or dock? Thanks

**Comment by trunkmaster** [comment]
Hi, here are my explanations:
1. Hm... For some reason it stops working (as well as autocomplition with Esc key). Could you please file an issue about this? I'll fix it ASAP.
2. You have 2 options for this:
   - open Login Window Prefererences (double-click on clock/date icon in Dock) and select section and specify program/script in "Login Hook" field;
   - open ~/Library/Preferences/.NextSpace/Workspace with editor, add line "ComposerEnabled = YES;" and restart Workspace (Workspace has undocumented and not tested xcompmgr inside)
3. There's no official way to set icon as in WindowMaker - I don't like that WM behavior before Xorg standards. It's definitely a part of Workspace I have to improve. But you may try to edit `~/Library/Preferences/.NextSpace/WMWindowAttributes.plist` and add name of application and path to icon.
4. This is a default behavior of NextStep's Workspace Manager as well as macOS's Finder. Although you may create a app wrapper. Look at the attached wrappers examples.
[AppWrappers.zip](https://github.com/trunkmaster/nextspace/files/14733792/AppWrappers.zip)

I hope at least some of my answers help you.
